VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,188 @@
+package org.edx.mobile.discussions;
+
+import android.text.SpannableString;
+import android.text.TextUtils;
+import android.text.style.ClickableSpan;
+import android.view.View;
+import android.widget.TextView;
+
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.test.BaseTestCase;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertTrue;
+
+public class DiscussionTextUtilsTest extends BaseTestCase {
+    @Test
+    public void testSetAuthorAttributionText_AllCombinations() {
+        TextView textView = new TextView(context);
+        String endorsePrefix = "Endorsed";
+        String answerPrefix = "Marked as answer";
+
+        long now = System.currentTimeMillis();
+        Date creationDate = new Date();
```

Comment:
This should probably be set to some time before the current time, so as to have a meaningful elapsed time label.